### PR TITLE
Cherry-picked changes from pull request #1392.

### PR DESCRIPTION
**Pull request by davebx**
This should pull in all the toolshed-specific changes from #1392

### CODE REVIEW

File: lib/galaxy/webapps/tool_shed/api/repositories.py [hunk 1]
```diff
@@ -242,10 +240,30 @@ def get_repository_revision_install_info( self, trans, name, owner, changeset_re
             log.debug( debug_msg )
             return {}, {}, {}
 
+    @web.expose_api_anonymous
+    def get_installable_revisions( self, trans, **kwd ):
+        """
+        GET /api/repositories/get_unordered_installable_revisions
```
Comment (nsoranzo) [inline lib/galaxy/webapps/tool_shed/api/repositories.py]: s/unordered_//

Also few lines below.

**Comment by martenson** [comment]
thanks for splitting this up @davebx ! makes it much easier to review

**Comment by martenson** [comment]
:+1:

File: lib/galaxy/webapps/tool_shed/api/repositories.py [hunk 2]
```diff
@@ -81,14 +81,14 @@ def add_repository_registry_entry( self, trans, payload, **kwd ):
...
-        trans.app.repository_registry.add_entry( repository )
+        self.app.repository_registry.add_entry( repository )
         response_dict[ 'status' ] = 'ok'
         response_dict[ 'message' ] = 'Entries for repository %s owned by %s have been added to the Tool Shed repository registry.' \
             % ( name, owner )
         return response_dict
 
     @web.expose_api_anonymous
-    def get_ordered_installable_revisions( self, trans, name, owner, **kwd ):
+    def get_ordered_installable_revisions( self, trans, **kwd ):
```
Comment (martenson) [inline lib/galaxy/webapps/tool_shed/api/repositories.py:91]: @davebx this signature change is not reflected in the documentation and maybe breaks backward compatibility (i.e. in bioblend)
reported in http://dev.list.galaxyproject.org/Test-tool-shed-api-repositories-get-ordered-installable-revisions-not-working-td4669149.html

Comment (bgruening) [inline lib/galaxy/webapps/tool_shed/api/repositories.py:91]: @davebx you can also test this with the ansible-tools role https://github.com/galaxyproject/ansible-galaxy-tools
It is not possible to install tools from the TTS anymore.

Comment (martenson) [inline lib/galaxy/webapps/tool_shed/api/repositories.py:91]: created an issue for it: https://github.com/galaxyproject/galaxy/issues/1940

File: lib/galaxy/webapps/tool_shed/api/repositories.py [hunk 3]
```diff
@@ -242,10 +240,30 @@ def get_repository_revision_install_info( self, trans, name, owner, changeset_re
...
+        :param tsr_id: the encoded toolshed ID of the repository
+
+        Returns a list of lists of changesets, in the format [ [ 0, fbb391dc803c ], [ 1, 9d9ec4d9c03e ], [ 2, 9b5b20673b89 ], [ 3, e8c99ce51292 ] ].
+        """
+        # Example URL: http://localhost:9009/api/repositories/get_installable_revisions?tsr_id=9d37e53072ff9fa4
+        tsr_id = kwd.get( 'tsr_id', None )
+        if tsr_id is not None:
+            repository = suc.get_repository_in_tool_shed( self.app, tsr_id )
+        else:
+            error_message = "Error in the Tool Shed repositories API in get_ordered_installable_revisions: "
```
Comment (nsoranzo) [inline lib/galaxy/webapps/tool_shed/api/repositories.py:257]: s/get_ordered_installable_revisions/get_installable_revisions/

**Pull request merged**